src/detect-bsize.c

```diff
@@ -291,6 +293,38 @@ static int DetectBsizeSetup (DetectEngineCtx *de_ctx, Signature *s, const char *
     DetectBsizeData *bsz = DetectBsizeParse(sizestr);
     if (bsz == NULL)
         goto error;
+
+    SigMatch *pm = NULL;
+    pm = DetectGetLastSMFromLists(s, DETECT_CONTENT, -1);
```

Comment:
this takes the last content from any buffer/list, which isn't right. bsize should only check the contents of the buffer/list it is on.

Also, it checks just the last content, not all of them. See my suggested test additions below.